src/main/java/com/iwaconsolti/league/demo/controller/TeamController.java

```diff
@@ -0,0 +1,100 @@
+package com.iwaconsolti.league.demo.controller;
+
+import com.iwaconsolti.league.demo.dto.request.TeamCreateRequest;
+import com.iwaconsolti.league.demo.dto.request.TeamUpdateRequest;
+import com.iwaconsolti.league.demo.dto.response.PlayerResponse;
+import com.iwaconsolti.league.demo.dto.response.TeamResponse;
+import com.iwaconsolti.league.demo.dto.response.TeamSummaryResponse;
+import com.iwaconsolti.league.demo.model.Team;
+import com.iwaconsolti.league.demo.service.TeamService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/leagues/{leagueId}/teams")
+@Slf4j
+@RequiredArgsConstructor
+public class TeamController {
+
+    private final TeamService teamService;
+
+
+    @PostMapping
+    public ResponseEntity<TeamResponse> createTeam(
+            @PathVariable long leagueId,
+            @RequestBody TeamCreateRequest request) {
+        try {
+            log.info("Creating team in league {}", leagueId);
+            Team team = new Team();
+            team.setName(request.name());
```

Comment:
In this cases where you only populated some propeties of the entities, is better to create a custom constructor that only receive the value used, to avoid the usage of setters